shell/common/tests/shell_test_surface.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```

Comment:
Is it possible to remove this class (and thus shell_test_surface.h/cc) and just merge the `CreateTestSurface/CreateRenderingSurface` into `ShellTestPlatformView`?

Reply:
This is the common interface used by `ShellTestSurfaceGL` and `ShellTestSurfaceMetal`. Without this common interface, the ShellTestPlatformView will have to own two unique pointers to both ShellTestSurfaceGL and ShellTestSurfacePlatform view and then conditionally check for the availability of either on each call. Now, the base can own just one member and call CreateRenderingSurface on it without null checks everywhere.

Reply:
Can `ShellTestPlatformView` only own a unique pointer to `GPUSurfaceDelegate` (the parent class of both `ShellTestSurfaceMetalImpl` and `ShellTestSurfaceGL`), and directly implement `CreateRenderingSurface` based on that? We can discuss offline about this as I feel that Github comments don't have the bandwidth to resolve this efficiently.